apps/backend/src/app/modules/killswitch/killswitch.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Inject, Injectable, OnModuleInit } from '@nestjs/common';
+import { EXTENDED_PRISMA_SERVICE } from '../database/db.constants';
+import { ExtendedPrismaService } from '../database/prisma.extension';
+import { KillswitchType } from './killswitch.enum';
+import { Enum } from '@momentum/enum';
+
+export type Killswitches = Record<KillswitchType, boolean>;
+export const KILLSWITCH_CONFIG_ID = 'Killswitches';
+
+@Injectable()
+export class KillswitchService implements OnModuleInit {
+  constructor(
+    @Inject(EXTENDED_PRISMA_SERVICE) private readonly db: ExtendedPrismaService
+  ) {}
+
+  private switches: Killswitches;
+
+  public async onModuleInit() {
+    const dbResponse = await this.db.config.findUnique({
+      where: { id: KILLSWITCH_CONFIG_ID }
+    });
+
+    const storedSwitches = dbResponse?.value as Killswitches;
+    if (storedSwitches) {
+      this.validateKillswitches(storedSwitches);
```

Comment:
Oh, this returns bool but we're not using it. If `validateKillswitches` returns false, let's log an **error** like "Stored killswitches failed validation" and init everything as false like the below code.

Sorry, I know I wrote this, so sort of reviewing my own stuff here!